sqlglot/expressions.py

```diff
@@ -171,7 +170,7 @@ class CharacterSet(Expression):
 
 class CTE(Expression):
     token_type = TokenType.WITH
-    arg_types = {'this': True, 'expressions': True, 'recursive': False}
+    arg_types = {'this': False, 'expressions': True, 'recursive': False}
```

Comment:
@tobymao I'm a little concerned about this change. Had to do this because of:
```python
  File "/Users/izeigerman/github/sqlglot/tests/test_dialects.py", line 517, in test_spark
    transpile(
  File "/Users/izeigerman/github/sqlglot/sqlglot/__init__.py", line 24, in transpile
    for expression in parse(code, read, error_level=error_level)
  File "/Users/izeigerman/github/sqlglot/sqlglot/__init__.py", line 13, in parse
    return dialect.parse(code, **opts)
  File "/Users/izeigerman/github/sqlglot/sqlglot/dialects.py", line 17, in parse
    return self.parser(**opts).parse(self.tokenizer().tokenize(code), code)
  File "/Users/izeigerman/github/sqlglot/sqlglot/parser.py", line 198, in parse
    self.raise_error(str(e))
  File "/Users/izeigerman/github/sqlglot/sqlglot/parser.py", line 225, in raise_error
    raise self.error
sqlglot.errors.ParseError: Required keyword: 'this' missing for TokenType.WITH. Line 0, Col: 34.
WITH RECURSIVE T(N) AS (VALUES (1))
```

Reply:
ah a recursive cte... i guess a recursive cte doesn't have the name

Reply:
shouldn't the this be T(N) in this example?

Reply:
Yeah, I kind of thought it should be `T` or something. Here's the parsed tree:
```
[(CTE this: , expressions:
  (ALIAS this:
    (VALUES expressions:
      (TUPLE expressions: 1)), alias:
    (ANONYMOUS this: T, expressions:
      (COLUMN this: N, db: , table: , fields: ))), recursive: RECURSIVE)]
```
Does this look correct to you?

Reply:
Ah, seems like it actually expects `SELECT` and not `VALUES`.

Reply:
ah, this may be a case of a bad test

`
WITH RECURSIVE t(n) AS (
    VALUES (1)
  UNION ALL
    SELECT n+1 FROM t WHERE n < 100
)
SELECT sum(n) FROM t;`

is valid, my test was probably a partial

Reply:
makes sense now!